src/System.Management.Automation/engine/Interop/Windows/WNetGetConnection.cs

```diff
@@ -26,56 +26,53 @@ internal static int GetUNCForNetworkDrive(char drive, out string? uncPath)
 
             uint bufferSize = MAX_PATH;
 
-#if DEBUG
-            // In Debug mode buffer size is initially set to 3 and if additional buffer is required, the
-            // required buffer size is allocated and the WNetGetConnection API is executed with the newly
-            // allocated buffer size.
-            bufferSize = 3;
-#endif
-
             ReadOnlySpan<char> driveName = stackalloc char[] { drive, ':', '\0' };
             Span<char> uncBuffer = stackalloc char[(int)bufferSize];
-            int errorCode = ERROR_NO_NETWORK;
 
+            char[]? rentedArray = null;
             try
             {
-                errorCode = WNetGetConnection(driveName, uncBuffer, ref bufferSize);
-            }
-            catch (System.DllNotFoundException)
-            {
-                s_WNetApiNotAvailable = true;
-                return ERROR_NOT_SUPPORTED;
-            }
-
-            if (errorCode == ERROR_SUCCESS)
-            {
-                // exclude null terminator
-                uncPath = uncBuffer.Slice(0, (int)bufferSize - 1).ToString();
-            }
-            else if (errorCode == ERROR_MORE_DATA)
-            {
-                char[]? rentedArray = null;
-                try
+                while (true)
                 {
-                    uncBuffer = rentedArray = ArrayPool<char>.Shared.Rent((int)bufferSize);
-                    errorCode = WNetGetConnection(driveName, uncBuffer, ref bufferSize);
+                    int errorCode;
+                    try
+                    {
+                        errorCode = WNetGetConnection(driveName, uncBuffer, ref bufferSize);
+                    }
+                    catch (System.DllNotFoundException)
+                    {
+                        s_WNetApiNotAvailable = true;
+                        return ERROR_NOT_SUPPORTED;
+                    }
 
-                    if (errorCode == ERROR_SUCCESS)
+                    if (errorCode == ERROR_MORE_DATA)
                     {
-                        // exclude null terminator
-                        uncPath = uncBuffer.Slice(0, (int)bufferSize - 1).ToString();
+                        uncBuffer = rentedArray = ArrayPool<char>.Shared.Rent((int)bufferSize);
```

Comment:
I've updated the code to return the rented array in all cases, I've kept the loop for now but if others object I can change it again.

I've kept the loop because now it only calls the function once and shared the same logic for handling errors and how to read the output on success.

Reply:
I don't see you added the pattern I pointed.